backends/p4tools/modules/testgen/core/small_step/extern_stepper.cpp

```diff
@@ -392,12 +401,8 @@ const ExprStepper::ExternMethodImpls<ExprStepper> ExprStepper::CORE_EXTERN_METHO
          if (!SymbolicEnv::isSymbolicValue(advanceExpr)) {
              stepToSubexpr(advanceExpr, stepper.result, stepper.state,
                            [&externInfo](const Continuation::Parameter *v) {
-                               auto *clonedCall = externInfo.originalCall.clone();
-                               auto *arguments = clonedCall->arguments->clone();
-                               auto *arg = arguments->at(0)->clone();
-                               arg->expression = v->param;
-                               (*arguments)[0] = arg;
-                               clonedCall->arguments = arguments;
+                               const auto *clonedCall =
+                                   replaceCallArg(&externInfo.originalCall, 0, v->param);
```

Comment:
This is definitely an improvement and helps avoiding accidental bugs by getting an index or variable wrong.